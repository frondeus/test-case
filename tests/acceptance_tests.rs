#![cfg(test)]

mod acceptance {
    use itertools::Itertools;
    use std::env;
    use std::path::PathBuf;
    use std::process::Command;

    #[test]
    fn basic() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("basic"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn hamcrest_assertions() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("hamcrest_assertions"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn r#async() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("async"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }
}
