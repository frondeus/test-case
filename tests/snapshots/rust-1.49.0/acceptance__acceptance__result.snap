---
source: tests/acceptance_tests.rs
assertion_line: 114
expression: lines

---
    tests::is_even::_13_expects
    tests::is_odd_boxed::_12_expects
  left: `1`,
  left: `1`,
 right: `0`: the test returned a termination value with a non-zero status code (1) which indicates a failure', /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/test/src/lib.rs:193:5
 right: `0`: the test returned a termination value with a non-zero status code (1) which indicates a failure', /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/test/src/lib.rs:193:5
---- tests::is_even::_13_expects stdout ----
---- tests::is_odd_boxed::_12_expects stdout ----
Error: "is even"
Error: "is odd"
failures:
failures:
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
running 7 tests
test result: FAILED. 4 passed; 2 failed; 1 ignored; 0 measured; 0 filtered out
test tests::ignore_supported::_12_expects_inconclusivematching_ok_ ... ignored
test tests::is_even::_12_expects ... ok
test tests::is_even::_13_expects ... FAILED
test tests::is_odd_boxed::_12_expects ... FAILED
test tests::is_odd_boxed::_13_expects ... ok
test tests::panics_supported::_12_expects_panicking_none ... ok
test tests::panics_supported::_13_expects_panicking_some_with_text_ ... ok
thread 'tests::is_even::_13_expects' panicked at 'assertion failed: `(left == right)`
thread 'tests::is_odd_boxed::_12_expects' panicked at 'assertion failed: `(left == right)`
